main-application/comms-processing_tasks.c

```diff
@@ -0,0 +1,72 @@
+/*  comms-processing_tasks.c
+
+    Created by Minh Phong Bill Truong on April 28th 2020
+ */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "semphr.h"
+
+#include <stdio.h>
+#include "comms-processing_tasks.h"
+
+//==============================================================================
+//                                   GLOBALS
+//==============================================================================
+/**
+ * This Queue is used to store received and unprocessed packets
+ */
+QueueHandle_t rawPacketReceiveQueue;
+
+/**
+ * This Queue is used to store deparsed packets that will be used by transmitTask
+ */
+QueueHandle_t packetsToTransmit;
+
+//==============================================================================
+//                                  FUNCTIONS
+//==============================================================================
+
+/**
+ *  @brief Call functions required to deparse packet
+ *  
+ *  The deparser task receives the data that is to be sent down (clarify) and calls the
+ *  required functions to handle the formatting of the data into a packet. 
+ * 
+ *  @header	"software-and-command/main-applicatoion/comms-processing_tasks.c"
+ *	@param	Unused
+ *	@pre	None
+ * 	@post	Formatted packet is stored in memory
+ * 	@return	None
+ */
+void deparseTask(void* pvparameters) 
+{
+    uint16_t  rawPacket;
+    uint16_t  deparsedPacket;
+    while(1) 
+    {
+        /*
+        * Execute any tasks of equal priority that are Ready now.
+        */
+        taskYIELD();
```

Comment:
Interesting approach to always yielding execution immediately after entering the task, before running. Is there anything else that this task should be blocked by? i.e. any specific event group flags, or states? etc

Reply:
I believe this task should only be running if there are raw packets in the receive queue ( rawPacketReceiveQueue ) that need to be deparsed, otherwise the task should yield immediately. I think the implementation for this is to just make a conditional to check if there is anything to be deparsed. Thoughts?

Reply:
I would agree